@vmswr/plot.m

```diff
@@ -25,225 +24,226 @@
-				if( (obj.data.markers(n,3)==Args.RewardMarker) || (floor(obj.data.markers(n,3)/10)==Args.RewardMarker) )
-					% indicate correct trial
-					line(repmat((obj.data.analogTime(idx(end))-obj.data.analogTime(tIdx(1)))*1000,2,1),ylim,'Color','b')
-				else
-					% indicate incorrect trial
-					line(repmat((obj.data.analogTime(idx(end))-obj.data.analogTime(tIdx(1)))*1000,2,1),ylim,'Color','r')
-				end
-			end
-		end
-	end
 else
 	% plot all data
-	if(Args.FreqPlot)
-		sRate = obj.data.analogInfo.SampleRate;
-		data = obj.data.analogData;
-		if(~isempty(Args.RemoveLineNoise))
-			data = nptRemoveLineNoise(data,Args.RemoveLineNoise,sRate);
-		end
+	n = 1;
+    fprintf('work in progress\n');
+end
 
-		datam = mean(data);
-		PlotFFT(data-datam,sRate)
-		set(gca,'TickDir','out')
-		if(Args.LogPlot)
-			set(gca,'YScale','log')
-		end
-	else
-		if(size(obj.data.trialIndices,2)<3)
-			OldMarkerFormat = 1;
-		else
-			OldMarkerFormat = 0;
-		end
-		
-		if(OldMarkerFormat)
-			dIdx = diff(obj.data.trialIndices,1,2);
-		else
-			dIdx = obj.data.trialIndices(:,3) - obj.data.trialIndices(:,1);
-		end
-		% find longest trial
-		mIdx = max(dIdx);
-		% create matrix
-		mdata = zeros(obj.data.numSets,mIdx);
-		for i = 1:obj.data.numSets
-			idx = obj.data.trialIndices(i,:);
-			if(Args.NormalizeTrial)
-				if(OldMarkerFormat)
-					rdata = obj.data.analogData(idx(1):idx(2));
-				else
-					rdata = obj.data.analogData(idx(1):idx(3));
-				end
-				rdmin = min(rdata);
-				rdmax = max(rdata);
-				mdata(i,1:(dIdx(i)+1)) = (rdata-rdmin)/(rdmax-rdmin);
-			else
-				if(OldMarkerFormat)
-					mdata(i,1:(dIdx(i)+1)) = obj.data.analogData(idx(1):idx(2));
-				else
-					mdata(i,1:(dIdx(i)+1)) = obj.data.analogData(idx(1):idx(3));
-				end
-			end
-		end
-		imagesc(mdata)
-		colormap(jet)
-	end
+tIdx = obj.data.trialIndices(n,:);
+sRate = obj.data.analogInfo.SampleRate;
+swrInfo = obj.data.analogRmsInfo.Swr;
+swrMean = obj.data.analogRmsInfo.Mean;
+swrStd = obj.data.analogRmsInfo.Std;
+
+if(length(tIdx)==2)
+    OldMarkerFormat = 1;
+else
+    OldMarkerFormat = 0;
 end
 
-% add code for plot options here
-%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
-%
-% @vmlfp/PLOT takes 'LabelsOff' as an example
-if(~Args.LabelsOff)
-	if(Args.FreqPlot)
-		xlabel('Frequency (Hz)')
-		ylabel('Magnitude')
-	elseif(Args.TFfft)
-		xlabel('Time (s)')
-		ylabel('Frequency (Hz)')
-	else
-		xlabel('Time (ms)')
-		ylabel('Voltage (uV)')
-	end
+if(OldMarkerFormat)
+    idx = (tIdx(1)-(Args.PreTrial/1000*sRate)):tIdx(2);
+else
+    idx = (tIdx(1)-(Args.PreTrial/1000*sRate)):tIdx(3);
 end
-if(~Args.TitleOff)
-    [a,b] = fileparts(obj.nptdata.SessionDirs{:});
-    title(b)
+
+if(Args.Rms)
+    idx=unique(roundn(idx./5,0));
+    if (Args.PlotAllData)
+        data = obj.data.analogRmsData';
+    else
+        data = obj.data.analogRmsData';
+        data = data(idx,:);
+    end
+    sc = 200; %scaling for graphs
```

Comment:
Please avoid using constants in your code. Set them up as optional input arguments instead.